client/app/bundles/course/survey/components/QuestionForm.jsx

```diff
@@ -0,0 +1,296 @@
+    const { disabled } = this.props;
+    return (
+      <Field
+        component={TextField}
+        type="number"
+        style={styles.numberOfResponsesField}
+        inputStyle={styles.numberOfResponsesInput}
+        underlineStyle={styles.numberOfResponsesInput}
+        {...{ name, floatingLabelText, disabled }}
+      />
+    );
+  }
+
+  renderSpecificFields(questionType) {
+    const { intl, disabled } = this.props;
+    const { TEXT, MULTIPLE_CHOICE, MULTIPLE_RESPONSE } = questionTypes;
+
+    let specificFields;
+    switch (questionType) {
+      case MULTIPLE_CHOICE:
+        specificFields = (
+          <FieldArray name="options" component={this.renderMcqOptions} {...{ disabled }} />
+        );
+        break;
+      case MULTIPLE_RESPONSE:
+        specificFields = (
+          <div>
+            <div style={styles.numberOfResponsesDiv}>
+              {this.renderNumberOfResponsesField(
+                'min_options',
+                intl.formatMessage(translations.minOptions)
+              )}
+              {this.renderNumberOfResponsesField(
+                'max_options',
+                intl.formatMessage(translations.maxOptions)
+              )}
+            </div>
+            <FieldArray name="options" component={this.renderMrqOptions} {...{ disabled }} />
+          </div>
+        );
+        break;
+      case TEXT:
+      default:
+        specificFields = [];
+    }
+    return specificFields;
+  }
+
+  render() {
+    const { handleSubmit, intl, onSubmit, disabled, formValues } = this.props;
+    const { TEXT, MULTIPLE_CHOICE, MULTIPLE_RESPONSE } = questionTypes;
+    const questionType = formValues && formValues.question_type;
+
+    return (
+      <Form onSubmit={handleSubmit(onSubmit)}>
+        <div style={styles.fieldRow}>
+          <Field
+            name="question_type"
+            floatingLabelText={intl.formatMessage(translations.questionType)}
+            component={SelectField}
+            style={styles.questionType}
+            {...{ disabled }}
+          >
+            <MenuItem
+              value={TEXT}
+              primaryText={intl.formatMessage(translations.textResponse)}
+            />
+            <MenuItem
+              value={MULTIPLE_CHOICE}
+              primaryText={intl.formatMessage(translations.multipleChoice)}
+            />
+            <MenuItem
+              value={MULTIPLE_RESPONSE}
+              primaryText={intl.formatMessage(translations.multipleResponse)}
+            />
+          </Field>
+          <Field
+            name="required"
+            label={intl.formatMessage(questionFormTranslations.compulsory)}
+            component={Toggle}
+            style={styles.requiredToggle}
+            labelStyle={styles.requiredToggleLabel}
+            {...{ disabled }}
+          />
+        </div>
+        <Field
+          name="description"
+          floatingLabelText={intl.formatMessage(translations.question)}
```

Comment:
So this is the "content" of the question?  I am asking because it actually confuses me: There is a "question" under "new question", I am not very clear what does that "question" mean.

Maybe we should use "question text" instead ?